core/src/main/java/org/infinispan/expiration/ExpirationManager.java

```diff
@@ -29,7 +30,17 @@
    void processExpiration();
 
    /**
-    * @return true if expiration is enabled, false otherwise
+    * @return true if expiration reaper is enabled, false otherwise
     */
    boolean isEnabled();
+
+   /**
+    * This should be invoked passing in an entry that is now expired.  Note this method is assumed to not have the
```

Comment:
s/is assumed/assumes/

Also, I'm not sure this will work well with Pedro's locking PR https://github.com/infinispan/infinispan/pull/3606, which changes the lock owner from the calling thread to the command's invocation id.